internal/relation/endpoint.go

```diff
@@ -74,3 +75,36 @@ func NaturalKey(endpoints []Endpoint) corerelation.Key {
 	}
 	return corerelation.Key(strings.Join(endpointNames, " "))
 }
+
+// EndpointIdentifier is the natural key of a relation endpoint.
+type EndpointIdentifier struct {
+	// ApplicationName is the name of the application the endpoint belongs to.
+	ApplicationName string
+	// EndpointName is the name of the endpoint.
+	EndpointName string
+}
+
+// ParseRelationKey parses a relation key into endpointIdentifiers. It expects a
+// key of one of the following forms:
+// - "<application-name>:<endpoint-name> <application-name>:<endpoint-name>"
+// - "<application-name>:<endpoint-name>"
+func ParseRelationKey(relationKey corerelation.Key) ([]EndpointIdentifier, error) {
```

Comment:
This is better off in core or the domain.  It's an accessor type. See prior comment.